src/app_service/service/accounts/service.nim

```diff
@@ -420,13 +260,58 @@ QtObject:
       error "failed to create account or login", procName="createAccountAndLogin", errName = e.name, errDesription = e.msg
       return e.msg
 
-  proc importAccountAndLogin*(self: Service, mnemonic: string, password: string, recoverAccount: bool, displayName: string, imagePath: string, imageCropRectangle: ImageCropRectangle): string =
+  proc importAccountAndLogin*(self: Service, 
+    mnemonic: string, 
+    password: string, 
+    recoverAccount: bool, 
+    displayName: string,
+    imagePath: string, 
+    imageCropRectangle: ImageCropRectangle,
+    keycardInstanceUID: string = "",
+  ): string =
+
+    var request = RestoreAccountRequest(
+      mnemonic: mnemonic,
+      fetchBackup: recoverAccount,
+      createAccountRequest: self.buildCreateAccountRequest(password, displayName, imagePath, imageCropRectangle),
+    )
+    request.createAccountRequest.keycardInstanceUID = keycardInstanceUID
+      
+    self.restoreAccountAndLogin(request)
+
+  # TODO: refactor arguments list
```

Comment:
Still relevant?

Reply:
nah, I guess it's fine to keep as is. Not a big win to refactor this.